databricks_cli/secrets/cli.py

```diff
@@ -0,0 +1,313 @@
+              ' assigned to the supplied principal (user or group). Currently, the only supported'
+              ' principal for this option is the group "users", which contains all users in the'
+              ' workspace. If not specified, the initial ACL with MANAGE permission applied to the'
+              ' scope is assigned to the request issuer\'s user identity.')
+@profile_option
+@eat_exceptions
+@provide_api_client
+def create_scope(api_client, scope, initial_manage_principal):
+    """
+    Creates a new secret scope with given name.
+    """
+    SecretApi(api_client).create_scope(scope, initial_manage_principal)
+
+
+def _scopes_to_table(scopes_json):
+    ret = []
+    for s in scopes_json.get('scopes', []):
+        ret.append((truncate_string(s['name']), s['backend_type']))
+    return ret
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help='Lists all secret scopes.')
+@click.option('--output', help=OutputClickType.help, type=OutputClickType())
+@profile_option
+@eat_exceptions
+@provide_api_client
+def list_scopes(api_client, output):
+    """
+    Lists all secret scopes.
+    """
+    scopes_json = SecretApi(api_client).list_scopes()
+    if OutputClickType.is_json(output):
+        click.echo(pretty_format(scopes_json))
+    else:
+        click.echo(tabulate(_scopes_to_table(scopes_json), headers=SCOPE_HEADER))
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help='Deletes a secret scope.')
+@click.option('--scope', required=True, type=SecretScopeClickType(), help=SecretScopeClickType.help)
+@profile_option
+@eat_exceptions
+@provide_api_client
+def delete_scope(api_client, scope):
+    """
+    Deletes a secret scope.
+    """
+    SecretApi(api_client).delete_scope(scope)
+
+
+def _verify_and_translate_options(string_value, binary_file):
+    """
+    Translates options into actual parameters for API call.
+    Return tuple with two values representing (string_value, bytes_value).
+    """
+    if string_value and binary_file:
+        error_and_quit("At most one of {} should be provided."
+                       .format(['string-value', 'binary-file']))
+
+    elif string_value is None and binary_file is None:
+        prompt = '# Do not edit the above line. Everything below it will be ignored.\n' + \
+            '# Please input your secret value above the line. Text will be stored in\n' + \
+            '# UTF-8 (MB4) form and any trailing new line will be stripped.\n' + \
+            '# Exit without saving will abort writing secret.'
+
+        # underlying edit function made sure using a temporary file for editing
+        content = click.edit('\n\n' + DASH_MARKER + prompt)
+        # return None means editor is closed without changes
+        if content is None:
+            error_and_quit('No changes made, write secret aborted.'
+                           ' Please follow the instruction to input secret value.')
+
+        elif DASH_MARKER not in content:
```

Comment:
I find it a bit weird that a user can't immediately delete everything in the buffer to add their secret. However, the example in http://click.pocoo.org/5/utils/#launching-editors is also weird in that accidentally editing the commented text will include it in the saved data.

I'm ok as is, since it's harder for a user to make a mistake and unexpected include unwanted data in the secret. :)